optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/bi/BiConstraintStream.java

```diff
@@ -771,6 +771,28 @@
      */
     <ResultA_> UniConstraintStream<ResultA_> map(BiFunction<A, B, ResultA_> mapping);
 
+    /**
+     * Takes each tuple and applies a mapping on the last fact, which turns it into {@link Iterable}.
+     * Returns a constraint stream consisting of tuples of the first fact
+     * and the contents of the {@link Iterable} one after another.
+     * This may produce a stream with duplicate tuples.
+     * See {@link #distinct()} for details.
```

Comment:
I suck at explaining things to humans. Suggestions welcome.

Reply:
For each tuple, does it create a cartesian product of `A` and items of `Iterable<ResultB_>` ?